app/models/season/phase_switcher.rb

```diff
@@ -1,44 +1,60 @@
+# frozen_string_literal: true
+
 class Season::PhaseSwitcher
 
+  PHASES = %i(
+    fake_proposals_phase
+    fake_application_phase
+    fake_coding_phase
+    back_to_reality
+    ).freeze
+
   def self.season
     Season.current
   end
 
+  def self.destined(phase)
+    raise ArgumentError.new("#{phase} is not a valid phase") unless phase.in? PHASES
+    self.public_send(phase)
+  end
+
+  private
```

Comment:
It's a classic. Katrina Owen has given whole talks about refactoring a case statement :v:

And yes, for public class methods it's effectively the same. However, if you want to have private ones, only using `private` is not enough, if I recall correctly, there would be some macros like `def_private_method` or some like this...
Therefore when I want to have private methods (basically all of the time) I use `class << self`.

And for service objects: they can be anything as long as you call them "service object". Oftentimes, in a Rails context, the may for instance look like this:
```ruby
class Service
  def initialize(val:)
    @val = val
  end

  def call
    perform_complex_logic
  end

  private

  attr_reader :val

  def perform_complex_logic
    # do stuff
  end
end
```

But they can essentially just be anything that suits your business logic needs 😉